pallas-network/src/multiplexer.rs

```diff
@@ -63,112 +65,126 @@ pub struct Segment {
     }
 
-    #[cfg(windows)]
-    pub async fn connect_named_pipe(
-        pipe_name: impl AsRef<std::ffi::OsStr>,
-    ) -> Result<Self, tokio::io::Error> {
-        let client = tokio::net::windows::named_pipe::ClientOptions::new().open(&pipe_name)?;
-        Ok(Self::NamedPipe(client))
-    }
+    // #[cfg(windows)]
+    // pub fn connect_named_pipe(pipe_name: impl AsRef<std::ffi::OsStr>) ->
+    // IOResult<Self> {     let client =
+    // tokio::net::windows::named_pipe::ClientOptions::new().open(&pipe_name)?;
+    //     Ok(Self::NamedPipe(client))
+    // }
 
-    pub async fn readable(&mut self) -> tokio::io::Result<()> {
+    pub fn into_split(self) -> (BearerReadHalf, BearerWriteHalf) {
         match self {
-            Bearer::Tcp(x) => x.readable().await,
+            Bearer::Tcp(x) => {
+                let (r, w) = x.into_split();
+                (BearerReadHalf::Tcp(r), BearerWriteHalf::Tcp(w))
+            }
 
             #[cfg(unix)]
-            Bearer::Unix(x) => x.readable().await,
-
-            #[cfg(windows)]
-            Bearer::NamedPipe(x) => x.readable().await,
+            Bearer::Unix(x) => {
+                let (r, w) = x.into_split();
+                (BearerReadHalf::Unix(r), BearerWriteHalf::Unix(w))
+            }
         }
     }
+}
+
+pub enum BearerReadHalf {
+    Tcp(tcp::tcp::OwnedReadHalf),
 
-    fn try_read(&mut self, buf: &mut [u8]) -> tokio::io::Result<usize> {
+    #[cfg(unix)]
+    Unix(unix::unix::OwnedReadHalf),
+}
+
+impl BearerReadHalf {
+    async fn read_exact(&mut self, buf: &mut [u8]) -> IOResult<usize> {
         match self {
-            Bearer::Tcp(x) => x.try_read(buf),
+            BearerReadHalf::Tcp(x) => x.read_exact(buf).await,
 
             #[cfg(unix)]
-            Bearer::Unix(x) => x.try_read(buf),
-
-            #[cfg(windows)]
-            Bearer::NamedPipe(x) => x.try_read(buf),
+            BearerReadHalf::Unix(x) => x.read_exact(buf).await,
         }
     }
+}
+
+pub enum BearerWriteHalf {
+    Tcp(tcp::tcp::OwnedWriteHalf),
+
+    #[cfg(unix)]
+    Unix(unix::unix::OwnedWriteHalf),
+}
 
-    async fn write_all(&mut self, buf: &[u8]) -> tokio::io::Result<()> {
+impl BearerWriteHalf {
+    async fn write_all(&mut self, buf: &[u8]) -> IOResult<()> {
         match self {
-            Bearer::Tcp(x) => x.write_all(buf).await,
+            Self::Tcp(x) => x.write_all(buf).await,
 
             #[cfg(unix)]
-            Bearer::Unix(x) => x.write_all(buf).await,
-
-            #[cfg(windows)]
-            Bearer::NamedPipe(x) => x.write_all(buf).await,
+            Self::Unix(x) => x.write_all(buf).await,
         }
     }
 
-    async fn flush(&mut self) -> tokio::io::Result<()> {
+    async fn flush(&mut self) -> IOResult<()> {
         match self {
-            Bearer::Tcp(x) => x.flush().await,
+            Self::Tcp(x) => x.flush().await,
 
             #[cfg(unix)]
-            Bearer::Unix(x) => x.flush().await,
-
-            #[cfg(windows)]
```

Comment:
comment out instead of delete like the other windows stuff?

Reply:
yes, good catch